software/ChromeAndPuppeteer.md

```diff
@@ -0,0 +1,135 @@
+    http://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/qt5-qtbase-common-$QTVER.el9.noarch.rpm \
+    http://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/qt5-qtbase-gui-$QTVER.el9.aarch64.rpm \
+    http://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/glx-utils-8.4.0-12.20210504git0f9e7d9.el9.aarch64.rpm \
+    http://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/pipewire-libs-0.3.47-2.el9.aarch64.rpm \
+    http://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/fdk-aac-free-2.0.0-8.el9.aarch64.rpm \
+    http://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/libldac-2.0.2.3-10.el9.aarch64.rpm \
+    https://kojipkgs.fedoraproject.org//packages/chromium/$CHROMEVER/1.el8/aarch64/chromium-$CHROMEVER-1.el8.aarch64.rpm \
+    https://kojipkgs.fedoraproject.org//packages/chromium/$CHROMEVER/1.el8/aarch64/chromium-common-$CHROMEVER-1.el8.aarch64.rpm \
+    https://kojipkgs.fedoraproject.org//packages/chromium/$CHROMEVER/1.el8/aarch64/chromium-headless-$CHROMEVER-1.el8.aarch64.rpm \
+    https://kojipkgs.fedoraproject.org//packages/chromium/$CHROMEVER/1.el8/aarch64/chromedriver-$CHROMEVER-1.el8.aarch64.rpm
+```
+As QT version above will be updated, and the old one become unavailable at some point, the version variable will need to be changed accordingly.
+If this is the case, check: https://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/ to see which version is available.
+```
+rm .cache/puppeteer/chrome/linux-116.0.5845.96/chrome-linux64/chrome
+ln -s /usr/bin/chromium-browser .cache/puppeteer/chrome/linux-116.0.5845.96/chrome-linux64/chrome
+```
+This brute force install works because:
+```
+/snap/bin/chromium --version
+Chromium 116.0.5845.96 snap
+```
+Puppeteer needs a particular version of chromium and in this case this works.
+Version puppeteer(-core) 21.1.0 uses Chromium 116.0.5845.96
+
+### Code Examples
+
+Example code for Puppeteer can be found at https://github.com/puppeteer/examples
+
+### Virtual Framebuffer Xserver
+
+Some code examples, such as puppeteer/examples/oopif.js, may need a 'headful' chrome and thus an Xserver.
+A virtual framebuffer Xserver can be used for that.
+
+Ubuntu-22: ```sudo apt install xvfb```
+AL2023: ```sudo yum install Xvfb```
+```
+Xvfb &
+export DISPLAY=:0
+```
+When Chrome is now invoked in headful mode, it has an Xserver to render to.
+
+This can be tested with:
+```
+node puppeteer/examples/oopif.js
+```
+The oopif.js example invokes chrome in headful mode.
+
+
+## Appendix.
+
+### Code example to show the difference in API between PhantomJS and Puppeteer.
+
+
+Puppeteer Screenshot:
+```
+'use strict';
+
+const puppeteer = require('puppeteer');
+
+(async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  await page.goto('http://example.com', {waitUntil: 'load', timeout: 200});
```

Comment:
Does this work if you change this to http://www.google.com ?  That way its the same example.